core/templates/pages/contributor-dashboard-page/opportunities-list/opportunities-list.component.ts

```diff
@@ -20,17 +20,92 @@ require(
   'pages/contributor-dashboard-page/opportunities-list-item/' +
   'opportunities-list-item.component.ts');
 
+import { Subscription } from 'rxjs';
+
 angular.module('oppia').component('opportunitiesList', {
   bindings: {
-    opportunityDataIsLoading: '=',
-    getOpportunities: '&opportunities',
-    isMoreOpportunitiesAvailable: '&moreAvailable',
-    isLabelRequired: '&labelRequired',
-    isProgressBarRequired: '&progressBarRequired',
-    onLoadMoreOpportunities: '&',
-    onClickActionButton: '=',
-    getOpportunityHeadingTruncationLength:
-      '&opportunityHeadingTruncationLength'
+    loadOpportunities: '<',
+    labelRequired: '<',
+    progressBarRequired: '<',
+    loadMoreOpportunities: '<',
+    onClickActionButton: '<',
+    opportunityHeadingTruncationLength: '<'
   },
-  template: require('./opportunities-list.component.html')
+  template: require('./opportunities-list.component.html'),
+  controller: [
+    '$rootScope', 'ContributionOpportunitiesService',
+    'TranslationLanguageService', 'OPPORTUNITIES_PAGE_SIZE', function(
+        $rootScope, ContributionOpportunitiesService,
+        TranslationLanguageService, OPPORTUNITIES_PAGE_SIZE) {
+      var ctrl = this;
+      var opportunities = [];
+      ctrl.visibleOpportunities = [];
+      ctrl.directiveSubscriptions = new Subscription();
+
+      ctrl.directiveSubscriptions.add(
+        TranslationLanguageService.onActiveLanguageChanged.subscribe(
+          () => ctrl.$onInit()));
+
+      ctrl.directiveSubscriptions.add(
+        ContributionOpportunitiesService
+          .reloadOpportunitiesEventEmitter.subscribe(() => ctrl.$onInit()));
+
+      ctrl.directiveSubscriptions.add(
+        ContributionOpportunitiesService
+          .removeOpportunitiesEventEmitter.subscribe((opportunityIds) => {
+            opportunities = opportunities.filter(function(opportunity) {
+              return opportunityIds.indexOf(opportunity.id) < 0;
+            });
+            ctrl.visibleOpportunities = opportunities.slice(
+              0, OPPORTUNITIES_PAGE_SIZE);
+          }));
+
+      ctrl.$onDestroy = function() {
+        ctrl.directiveSubscriptions.unsubscribe();
+      };
+
+      ctrl.$onInit = function() {
+        opportunities = [];
+        ctrl.visibleOpportunities = [];
+        ctrl.activePageNumber = 1;
+        ctrl.lastPageNumber = 1000;
```

Comment:
Can you make these numbers constants, like min/max page number?

Reply:
This is not a constant, it's just a value to represent kinda infinite intially.